dvc/utils/studio.py

```diff
@@ -111,3 +115,52 @@ def env_to_config(env: dict[str, Any]) -> dict[str, Any]:
     if DVC_STUDIO_URL in env:
         config["url"] = env[DVC_STUDIO_URL]
     return config
+
+
+def get_subrepo_relpath(repo: "Repo") -> str:
+    from dvc.fs import GitFileSystem
+
+    scm_root_dir = "/" if isinstance(repo.fs, GitFileSystem) else repo.scm.root_dir
+
+    relpath = as_posix(repo.fs.relpath(repo.root_dir, scm_root_dir))
+
+    return "" if relpath == "." else relpath
+
+
+def get_dvc_experiment_parent_data(
+    repo: "Repo", baseline_rev: Union[str, None]
+) -> Union[dict[str, Any], None]:
+    from scmrepo.exceptions import SCMError
+
+    from dvc.scm import NoSCM
+
+    scm = repo.scm
+
+    if (
+        not baseline_rev
+        or not scm
+        or isinstance(scm, NoSCM)
+        or not (sha := scm.resolve_rev(baseline_rev))
+        or not (commit := scm.resolve_commit(sha))
+    ):
+        return None
+
+    dvc_experiment_parent_data = {
+        "sha": commit.hexsha,
+        "message": commit.message,
+        "title": commit.message.partition("\n")[0].strip(),
+        "author": {
+            "name": commit.author_name,
+            "email": commit.author_email,
+        },
+        "date": commit.commit_datetime.isoformat(),
+    }
+
+    try:
+        if branch := scm.active_branch():
```

Comment:
[I] Try and get this information on the Studio side.

Reply:
there is no point in getting this information from DVC because

1. There is "branch merging" logic in Studio
2. A lot (if not all) of the time HEAD is in a detached state without a branch.

Reply:
Add parent_shas and test that they are added in the correct order (newest to oldest). Mark behaviour test as studio dependency